src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx

```diff
@@ -1,51 +1,61 @@
-import React, { useEffect } from 'react';
-import IndexStore from '../index.store';
-import { useState } from 'react';
-import { Node } from '../appDetails.type';
+import React, { useEffect } from 'react'
+import IndexStore from '../index.store'
+import { useState } from 'react'
+import { Node } from '../appDetails.type'
 
 interface TabState {
-    status: string,
-    count: number,
+    status: string
+    count: number
     isSelected: boolean
 }
 
-export const StatusFilterButtonComponent = ({nodes}: {nodes: Array<Node>}) => {
-    const [selectedTab, setSelectedTab] = useState('all');
+export const StatusFilterButtonComponent = ({
+    nodes,
+    handleFilterClick,
+}: {
+    nodes: Array<Node>
+    handleFilterClick?: (selectedFilter: string) => void
+}) => {
+    const [selectedTab, setSelectedTab] = useState('all')
 
     let allNodeCount: number = 0,
         healthyNodeCount: number = 0,
         progressingNodeCount: number = 0,
-        failedNodeCount: number = 0;
+        failedNodeCount: number = 0
 
     nodes?.forEach((_node) => {
-        let _nodeHealth = _node.health?.status;
+        let _nodeHealth = _node.health?.status
 
         if (_nodeHealth?.toLowerCase() === 'healthy') {
-            healthyNodeCount++;
+            healthyNodeCount++
         } else if (_nodeHealth?.toLowerCase() === 'degraded') {
-            failedNodeCount++;
+            failedNodeCount++
         } else if (_nodeHealth?.toLowerCase() === 'progressing') {
-            progressingNodeCount++;
+            progressingNodeCount++
         }
-        allNodeCount++;
-    });
+        allNodeCount++
+    })
 
     const filters = [
-        { status: 'ALL', count: allNodeCount, isSelected: "all" == selectedTab },
-        { status: 'HEALTHY', count: healthyNodeCount, isSelected: "healthy" == selectedTab },
-        { status: 'DEGRADED', count: failedNodeCount, isSelected: "degraded" == selectedTab },
-        { status: 'PROGRESSING', count: progressingNodeCount, isSelected: "progressing" == selectedTab },
-    ];
+        { status: 'ALL', count: allNodeCount, isSelected: 'all' == selectedTab },
+        { status: 'HEALTHY', count: healthyNodeCount, isSelected: 'healthy' == selectedTab },
+        { status: 'DEGRADED', count: failedNodeCount, isSelected: 'degraded' == selectedTab },
+        { status: 'PROGRESSING', count: progressingNodeCount, isSelected: 'progressing' == selectedTab },
+    ]
 
     // const handleFilterClick = (filterName: string) => {
     //     IndexStore.updateFilterType(filterName);
     //     setSelectedTab(filterName.toLowerCase());
     // };
 
-    useEffect(()=>{
+    useEffect(() => {
         // handleFilterClick(selectedTab.toUpperCase())
```

Comment:
if not in use we can remove this commented code

Reply:
Done